dataset viewer on private datasets (#1206)
huggingface/hub-docs: docs/hub/datasets-viewer.md

To have a properly working Dataset Viewer for your dataset, make sure your dataset is in a supported format and structure.
There is also an option to configure your dataset using YAML.

> For private datasets, the Dataset Viewer is a feature available to [PRO users](https://huggingface.co/pricing) and [Enterprise organizations](https://huggingface.co/enterprise).

Reviewer: ```suggestion
For **private** datasets, the Dataset Viewer is enabled for [PRO users](https://huggingface.co/pricing) and [Enterprise Hub organizations](https://huggingface.co/enterprise).
```
